libs/blockchain/src/blockchain-rpc-query.service.ts

```diff
@@ -467,6 +467,7 @@ export class BlockchainRpcQueryService implements OnApplicationBootstrap, OnAppl
       };
     }
 
+    this.logger.error(`Expected HandleClaimed event but found ${event}`);
```

Comment:
Was going to throw error but it didn't help since it would cause the block scanning process to be stuck on that certain block and would not be able to recover. The better option seemed like to only log these cases. Open to other suggestions.